### DISCUSSION TITLE
Sample constraints in goldfish

### DISCUSSION THREAD

**Discussion by joseluisesna** [stocnet/goldfish#94] [General]
Hi, I would like to ask whether goldfish allows for sample constraints in a similar fashion as ERGM (see section 2: https://statnet.org/workshop-advanced-ergm/advanced_ergm_tutorial.html). From what I understood, if I wanted to set sample constraints in who can or cannot send a tie, turning the present feature to TRUE or FALSE will account for this. However, do you happen to have implemented sample space constraints for the choice function (so actors do not consider certain others as potential recipients of their ties)?

**Comment by auzaheta** [comment]
Hi @joseluisesna,

I want to start by clarifying some terminology. In the REM context, the term used by [Marcum and Butts](https://www.jstatsoft.org/article/view/v064i05) to refer to the set of ties that are allowed to create events, is support constraint, as opposed to the sample space constraint definition in ERGM. The support constraint determines which pairs $(i,j)$ are allowed to compete to create an event, whereas the sample space constraint limits the possible networks that will be included in the computation of the normalization constant (I'm not even sure how to explain the one from ERGM). 

Additionally, it's important to highlight that the constraint given by the `present` variable applies to both the set of possible senders and the choice set (the set of possible receivers).

Regarding your question about implementing support constraints for the choice function in the goldfish package, where actors do not consider certain others as potential recipients, @marion-hoffman has implemented this feature. 

This functionality is available through the `opportunitiesList` parameter, which can be input in the `estimationInit` argument of the `estimate()` function. The opportunitiesList should be a list with the same length as the dependent events. 
For each event, you should provide an integer vector with the indices of the actors to include in the choice set. The indices correspond to the row numbers of the nodes dataset at the moment of processing. Please note that this feature currently only works with the default engine. 

You can see an example of the use of this functionality on the [DyNAM-i vignette](https://stocnet.github.io/goldfish/articles/dynami-example.html).
